AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreSyncExpression.swift

```diff
@@ -14,6 +14,25 @@ public struct DataStoreSyncExpression {
     let modelSchema: ModelSchema
     let modelPredicate: QueryPredicateResolver
 
+    init(modelSchema: ModelSchema, modelPredicate syncPredicate: @escaping QueryPredicateResolver) {
+        self.modelSchema = modelSchema
+        self.modelPredicate = {
+            let predicate = syncPredicate()
+            // Wrapping the predicate with a group AND enables
+            // AppSync to optimize the request by performing a DynamoDB query instead of a scan.
+            // If the provided syncPredicate is already a QueryPredicateGroup, this is not needed.
+            // If the provided group is of type AND, the optimization will occur.
+            // If the top level group is OR or NOT, the optimization is not possible anyway.
+            if predicate as? QueryPredicateGroup != nil {
+                return predicate
+            } else if let predicate = predicate as? QueryPredicateConstant,
+                      predicate == .all {
+                return predicate
```

Comment:
Will AppSync auto-optimize in the constant case? Is it possible or desirable to optimize by wrapping even constants in an `.and` group?